### PR TITLE
Add experiments to Dataproc notebook

### PR DESCRIPTION
**Pull request by wintwoo**
**REQUIRED:** Add a summary of your PR here, typically including why the change is needed and what was changed. Include any design alternatives for discussion purposes.

<br>
- Submit pipeline job with experiment parameter.<br>
- Add instructions for querying experiments from pipeline runs.<br>
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb [hunk 1]
```diff
@@ -98,7 +98,7 @@
       "source": [
```
Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>Random-forest&nbsp;-&gt; random-forest</p><p>Pyspark -&gt; PySpark</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>you would read -&gt; you read</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>Artefact -&gt; Artifact</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>Random-forest&nbsp;-&gt; random-forest&nbsp;</p><p>GCS -&gt; Cloud Storage</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>For links, use the style</p><p>Learn more about [title](link)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='62'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>Google Cloud Build -&gt; Cloud Build</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='68'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>combine the 2 text cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='70'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>dataset_uri not used?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='74'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <h4>Model Evaluation custom component -&gt;</h4><h4>Create model evaluation custom component</h4><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='75'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>dataset_uri not used?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='78'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>for links, use style</p><p>Learn more about [title](link)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='81'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='85'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>Add heading</p><p>GCS -&gt; Cloud Storage</p><p>link style</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='87'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <h3>Kubeflow Pipelines -&gt; Vertex AI Pipelines</h3>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='89'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>combine 2 text cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='99'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='100'/>

Comment (andrewferlitsch) [inline notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>Add heading.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='102'/>

Comment (wintwoo) [reply notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>I've removed the parameter. The Dataset artifact doesn't seem to be used at all, and the lineage it generates isn't useful in this pipeline.</p>

Comment (wintwoo) [reply notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb:98]: <p>I've removed the parameter. The Dataset artifact doesn't seem to be used at all, and the lineage it generates isn't useful in this pipeline.</p>

**Comment by wintwoo** [comment]
Need to investigate why the test is failing. The last step of the notebook cleans up some local directories by executing `rm -rf ...`, but the following (seemingly unrelated) error is thrown:

```
E1208 04:14:44.062827298   23155 backup_poller.cc:136]                 Run client channel backup poller: UNKNOWN:pollset_work {created_time:"2022-12-08T04:14:44.06263156+00:00", children:[UNKNOWN:Bad file descriptor {created_time:"2022-12-08T04:14:44.062590641+00:00", errno:9, os_error:"Bad file descriptor", syscall:"epoll_wait"}]}
```

**Comment by wintwoo** [comment]
@andrewferlitsch the notebook execution test appears to be timing out. The notebook includes cells that build container images, and the pipeline itself takes awhile to run due to multiple Spark jobs. These all take awhile to run.

From https://github.com/GoogleCloudPlatform/vertex-ai-samples/pull/1262/checks?check_run_id=9958151966

```
Step #5: notebooks/official/pipelines/google_cloud_pipeline_components_dataproc_tabular.ipynb FAILED in 00:14:55: Operation did not complete within the designated timeout of 900 seconds.
```

Can you suggest any ways to fix this - can the execution deadline be increased?

For some reason, this didn't when the last PR was submitted.

**Comment by wintwoo** [comment]
I thought the timeout originated from the test, sorry. I'll modify the
timeout value and try again.

**Comment by wintwoo** [comment]
@gericdong in the meantime, I downloaded the executed notebook from the test. I found the cell that executes `gcloud builds submit --tag $RUNTIME_CONTAINER_IMAGE $BUILD_PATH --machine-type=N1_HIGHCPU_32 --timeout=900s --verbosity=info` completes successfully. A new container image is pushed to Artifact Registry as expected:

```
<snip>

ID                                    CREATE_TIME                DURATION  SOURCE                                                                                                   IMAGES                                                                     STATUS
51d573d4-cdc3-4c5d-8f60-76848eabc6f6  2022-12-17T02:39:24+00:00  8M54S     gs://python-docs-samples-tests_cloudbuild/source/1671244758.935373-469aeec9a149403b885b0e4ae81b035c.tgz  gcr.io/python-docs-samples-tests/dataproc_serverless_custom_runtime:1.0.0  SUCCESS
```

The cells following this step also run without any apparent issues. Therefore, I'm not certain that the `gcloud builds submit` step  is causing issues (it hasn't been a problem before at least in previous PRs). However, I'll give it a try once I get the chance.

**Comment by ivanmkc** [comment]
@wintwoo please fetch and rebase on main to get fixes to the `vertex-ai-notebook-execution-test` check.

**Review by andrewferlitsch — APPROVED** [review]

**Comment by wintwoo** [comment]
@ivanmkc thanks, the test passed. Can you please merge?

**Pull request merged**